backend/src/LibExecution/Interpreter.fs

```diff
@@ -652,21 +652,24 @@ let rec eval'
 
 
     | EIf(id, cond, thenbody, elsebody) ->
+      let elsebodyUnwrapped = Option.defaultValue (EUnit id) elsebody
       match! eval state tat st cond with
       | DBool false ->
         do! preview tat st thenbody
-        return! eval state tat st elsebody
+        match elsebody with
+        | None -> return DUnit
+        | Some eb -> return! eval state tat st eb
       | DBool true ->
         let! result = eval state tat st thenbody
-        do! preview tat st elsebody
+        do! preview tat st elsebodyUnwrapped
         return result
       | cond when Dval.isFake cond ->
         do! preview tat st thenbody
-        do! preview tat st elsebody
+        do! preview tat st elsebodyUnwrapped
```

Comment:
Same for this branch.

Reply:
And here, neither the `thenBody` or `elseBody` seem appropriate since we're dealing with a fake val - we should probably return something else - not sure what

Reply:
We still want to preview both thenBody and elseBody here (unless there is no elseBody).